pkg/cloud/cloud.go

```diff
@@ -57,10 +58,14 @@ func GetCloudProvider(
 		)
 	case "GCP":
 		return gcp.NewGCPProvider(
+			restConfig,
 			kubeClient,
+			dynamicClient,
+			hubKubeClient,
 			eventsRecorder,
-			infraId, clusterName, config.Spec.CredentialsSecret.Name,
-			config.Spec.IPSecNATTPort, config.Spec.NATTDiscoveryPort,
+			region, infraId, clusterName, config.Spec.CredentialsSecret.Name,
+			config.Spec.GatewayConfig.AWS.InstanceType,
```

Comment:
Using "" now , we don't have GCP instance type now accepted from the UI, once that is added we can pass that.

Reply:
Please note that we need the GatewayInstanceType only when we are interested in creating dedicated gateway nodes, otherwise existing worker nodes will be labelled. So, you can just pass "" for now since you are not really creating dedicated gateway nodes.